File: backup/Twitch/app/src/main/java/com/exact/twitch/ui/pagers/MediaPagerAdapter.java
package com.exact.twitch.ui.pagers;

import android.content.Context;
import androidx.annotation.Nullable;
import androidx.fragment.app.FragmentManager;

import com.exact.twitch.R;

public abstract class MediaPagerAdapter extends ItemAwareFragmentPagerAdapter {

    private final Context context;

    public MediaPagerAdapter(Context context, FragmentManager fm) {
        super(fm);
        this.context = context;
    }

    @Nullable
    @Override
    public CharSequence getPageTitle(int position) {
        int res = 0;
        switch (position) {
            case 0:
                res = R.string.live;
                break;
            case 1:
                res = R.string.videos;
                break;
            case 2:
                res = R.string.clips;
                break;
        }
        return context.getString(res);
    }

    @Override
    public int getCount() {
        return 3;
    }
}

File: app/src/main/java/com/github/exact7/xtra/ui/common/pagers/MediaPagerAdapter.kt
package com.github.exact7.xtra.ui.common.pagers

import android.content.Context
import androidx.fragment.app.FragmentManager

import com.github.exact7.xtra.R

abstract class MediaPagerAdapter(private val context: Context, fm: FragmentManager) : ItemAwareFragmentPagerAdapter(fm) {

    override fun getPageTitle(position: Int): CharSequence? {
        val id = when (position) {
            0 -> R.string.live
            1 -> R.string.videos
            2 -> R.string.clips
            else -> 0
        }
        return context.getString(id)
    }

    override fun getCount(): Int = 3
}
